packages/auth/android/src/main/java/io/invertase/firebase/auth/ReactNativeFirebaseAuthModule.java

```diff
@@ -995,14 +997,15 @@ public void verifyPhoneNumber(
     }
 
     // Reset the credential
-    mCredential = null;
+    String newCode = URLEncoder.encode(phoneNumber);
+    String  key = "RNFATOKEN" + newCode;
+    mPhoneAutoVerifyCredentials.remove(phoneNumber);
```

Comment:
I'm a bit confused about this line - when does `mPhoneAutoVerifyCredentials` get added with a credential with the phone number? It gets added with a `key` (String + Phone number), so this seems it'd never do anything?

Reply:
Also a more simpler way to do thing could be `credential.hashCode().toString()` and storing by the hash code?